connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/ArtifactManager.scala

```diff
@@ -51,13 +51,14 @@ import org.apache.spark.util.{SparkFileUtils, SparkThreadUtils}
 class ArtifactManager(
     userContext: proto.UserContext,
     sessionId: String,
-    channel: ManagedChannel) {
+    channel: ManagedChannel,
+    retryPolicy: GrpcRetryHandler.RetryPolicy) {
   // Using the midpoint recommendation of 32KiB for chunk size as specified in
   // https://github.com/grpc/grpc.github.io/issues/371.
   private val CHUNK_SIZE: Int = 32 * 1024
 
-  private[this] val stub = proto.SparkConnectServiceGrpc.newStub(channel)
-  private[this] val bstub = proto.SparkConnectServiceGrpc.newBlockingStub(channel)
+  private[this] val stub = new CustomSparkConnectStub(channel, retryPolicy)
+  private[this] val bstub = new CustomSparkConnectBlockingStub(channel, retryPolicy)
```

Comment:
Could we add functions that ArtifactManager to SparkConnectClient, and pass client here to the constructor, and let it use the client?

Reply:
Can do that!

Reply:
I am just not sure how much sense this makes from a design perspective with all those [addArtifact(s) methods](https://github.com/apache/spark/blob/ae3defcba23632a438ff646fcca13901cb53ec3f/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala#L207C8-L207C8) we currently have in the client. We could pass the stubs to the artifactManager from the client.

Reply:
I see. I was thinking of having the client class as a single narrow waist for all rpc communication, without analyzing too much whats in ArtifactManager, but looking how much is there, I think you're right.
Let's create the stubs in the client and pass these to ArtifactManager like you suggest.